### PR TITLE
Move "Enter a password" cmd instructions to shared/cmd

### PR DESCRIPTION
**Pull request by 0xKiwi**
Improvement for validator client

This PR moves the CMD "Enter password" to its own function under shared/cmd, it will be reused in a future PR so might as well relocate it.

Testing `terminal.ReadPassword` is rather complicated so I've made a tracking issue for it #5749

### CODE REVIEW

File: shared/cmd/helpers.go [hunk 1]
```diff
@@ -42,3 +43,16 @@ func ConfirmAction(actionText string, deniedText string) (bool, error) {
 
 	return confirmed, nil
 }
+
+func EnterPassword() (string, error) {
```
Comment (terencechain) [inline shared/cmd/helpers.go:49]: Functional comment

Comment (terencechain) [inline shared/cmd/helpers.go:49]: This also is a good unit test candidate

Comment (0xKiwi) [reply shared/cmd/helpers.go:49]: Good point!

Comment (nisdas) [reply shared/cmd/helpers.go:49]: +1 unit test would be good for this

File: validator/accounts/account.go [hunk 2]
```diff
@@ -136,16 +136,13 @@ func Exists(keystorePath string) (bool, error) {
...
-		log.Info("Create a new validator account for eth2")
-		log.Info("Enter a password:")
-		bytePassword, err := terminal.ReadPassword(int(os.Stdin.Fd()))
+		log.Info("Creating a new validator account for eth2")
+		enteredPassphrase, err := cmd.EnterPassword()
 		if err != nil {
-			log.Fatalf("Could not read account password: %v", err)
-			return path, passphrase, err
+			log.Fatal(err)
+			return path, enteredPassphrase, err
```
Comment (nisdas) [inline validator/accounts/account.go:143]: no point of returning the result here, when you have a fatal log above

Comment (0xKiwi) [reply validator/accounts/account.go:143]: Yeah this is duplicated several times in the validator client, was gonna make another PR making sure it's either handled or returned, never both. Is that fine?

File: shared/cmd/helpers.go (hunk 1 above)
Comment (0xKiwi) [reply shared/cmd/helpers.go:50]: Writing a test for `terminal.ReadPassword` is rather complicated, made a tracking issue at #5749

File: shared/cmd/helpers.go [hunk 3]
```diff
@@ -42,3 +44,18 @@ func ConfirmAction(actionText string, deniedText string) (bool, error) {
 
 	return confirmed, nil
 }
+
+// EnterPassword queries the user for their password through the terminal, in order to make sure it is
+// not passed in a visible way to the terminal.
+// TODO(#5749) This function is untested, would be a good candidate for  unit test.
```
Comment (terencechain) [inline shared/cmd/helpers.go:50]:
```suggestion
// TODO(#5749) This function is untested, would be a good candidate for unit test.
```

Comment (0xKiwi) [reply shared/cmd/helpers.go]: Nice catch, fixed.

File: shared/cmd/helpers.go [hunk 4]
```diff
@@ -42,3 +44,18 @@ func ConfirmAction(actionText string, deniedText string) (bool, error) {
 
 	return confirmed, nil
 }
+
+// EnterPassword queries the user for their password through the terminal, in order to make sure it is
+// not passed in a visible way to the terminal.
+// TODO(#5749) This function is untested, would be a good candidate for a unit test.
```
Comment (terencechain) [inline shared/cmd/helpers.go:50]:
```suggestion
// TODO(#5749): This function is untested and should be tested.
```

**Pull request merged**